tests/tailor.js

```diff
@@ -687,4 +691,68 @@ describe('Tailor', () => {
         });
     });
 
+    it('should unzip the fragment response if it is compressed', (done) => {
+        nock('https://fragment')
+            .get('/1').reply(200, 'hello')
+            .defaultReplyHeaders({
+                'content-encoding': 'gzip',
+            })
+            .get('/2')
+            .reply(200, ()=> {
+                return zlib.gzipSync('GZIPPED');
+            });
+
+        mockTemplate
+            .returns(
+                '<fragment src="https://fragment/1"></fragment>' +
+                '<fragment src="https://fragment/2"></fragment>'
+            );
+
+        getResponse('http://localhost:8080/test').then((response) => {
+            assert.equal(response.body,
+                '<html>' +
+                '<head></head>' +
+                '<body>' +
+                '<script data-pipe>p.start(0)</script>' +
+                'hello' +
+                '<script data-pipe>p.end(0)</script>' +
+                '<script data-pipe>p.start(1)</script>' +
+                'GZIPPED' +
+                '<script data-pipe>p.end(1)</script>' +
+                '</body>' +
+                '</html>'
+            );
+            done();
+        });
+    });
+
+    it('should close the streams properly during unzping error', (done) => {
+        nock('https://fragment')
+            .defaultReplyHeaders({
+                'content-encoding': 'gzip',
+            })
+            .get('/2')
+            .reply(200, ()=> {
+                return new Error('GZIP Error');
```

Comment:
Is it correct handling of gzip error?
Seems like error on request object, not on gzip stream.
You can just pass normal string and set header to 'gzip', in that case unzip will fire an error 'incorrect header check'.

see an example:

```
describe('zlib stream error', () => {
  it('should handle error', done => {
    const sourceStream = new PassThrough();
    const destStream = new PassThrough();
    const unzipStream = zlib.createUnzip();

    sourceStream
      .on('error', () => console.log('source stream error'))
      .pipe(unzipStream)
      .on('error', err => {
        assert.equal(err.message, 'incorrect header check');
        done();
      })
      .pipe(destStream);

    sourceStream.end(Buffer.from('uncompressed content'));
  });
});

```